<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Davis J. McCarthy" />


<title>Differential expression across all donors</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.4/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
div.sourceCode {
  overflow-x: visible;
}
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fibroblast-clonality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential expression across all donors</h1>
<h4 class="author"><em>Davis J. McCarthy</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-08-17</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180807)</code> </summary></p>
<p>The command <code>set.seed(20180807)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/davismcc/fibroblast-clonality/tree/6b5f8c7f63ef4bdadbbcd677045c0fd4e3a53080" target="_blank">6b5f8c7</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  CITATION
    Untracked:  LICENSE
    Untracked:  code/analysis_for_garx.Rmd
    Untracked:  data/canopy/
    Untracked:  data/cell_assignment/
    Untracked:  data/de_analysis_FTv62/
    Untracked:  data/donor_info_070818.txt
    Untracked:  data/donor_neutrality.tsv
    Untracked:  data/fdr10.annot.txt.gz
    Untracked:  data/high-vs-low-exomes.v62.ft.alldonors-filt_lenient.all_filt_sites.vep_most_severe_csq.txt
    Untracked:  data/high-vs-low-exomes.v62.ft.filt_lenient-alldonors.txt.gz
    Untracked:  data/human_H_v5p2.rdata
    Untracked:  data/human_c2_v5p2.rdata
    Untracked:  data/human_c6_v5p2.rdata
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_qc_filt.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_all_with_qc_labels.rds
    Untracked:  data/sce_merged_donors_cardelino_donorid_unstim_qc_filt.rds
    Untracked:  data/sces/
    Untracked:  data/simulations/
    Untracked:  data/variance_components/
    Untracked:  docs/figure/
    Untracked:  figures/
    Untracked:  output/differential_expression/
    Untracked:  output/donor_specific/
    Untracked:  output/nvars_by_category_by_donor.tsv
    Untracked:  output/nvars_by_category_by_line.tsv
    Untracked:  output/variance_components/

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/dac4720b6f9158fa5bd9649d0227a20a39e0e0f0/analysis/differential_expression.Rmd" target="_blank">dac4720</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-17
</td>
<td style="text-align:left;">
Tidying up output
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/ef72f36a18ecb27025aad314fa1b50e06b39b246/analysis/differential_expression.Rmd" target="_blank">ef72f36</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-17
</td>
<td style="text-align:left;">
Fixing little bug
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/davismcc/fibroblast-clonality/blob/bc1dbc56906aa594933101e441c763d1b1d76f42/analysis/differential_expression.Rmd" target="_blank">bc1dbc5</a>
</td>
<td style="text-align:left;">
davismcc
</td>
<td style="text-align:left;">
2018-08-17
</td>
<td style="text-align:left;">
Adding analysis of differential expression results
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>Here, we will lok at differential expresion between clones across all donors at the gene and gene set levels.</p>
<div id="load-libraries-data-and-de-results" class="section level2">
<h2>Load libraries, data and DE results</h2>
<p>Load the genewise differential expression results produced with the edgeR quasi-likelihood F test and gene set enrichment results produced with camera.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">params &lt;-<span class="st"> </span><span class="kw">list</span>()
params<span class="op">$</span>callset &lt;-<span class="st"> &quot;filt_lenient.cell_coverage_sites&quot;</span>
<span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_c6_v5p2.rdata&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_H_v5p2.rdata&quot;</span>))
<span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/human_c2_v5p2.rdata&quot;</span>))

de_res &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">paste0</span>(<span class="st">&quot;data/de_analysis_FTv62/&quot;</span>,
                         params<span class="op">$</span>callset, 
                         <span class="st">&quot;.de_results_unstimulated_cells.rds&quot;</span>))</code></pre></div>
<p>Load SingleCellExpression objects with data used for differential expression analyses.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fls &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;data/sces&quot;</span>)
fls &lt;-<span class="st"> </span>fls[<span class="kw">grepl</span>(params<span class="op">$</span>callset, fls)]
donors &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.*ce_([a-z]+)_.*&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">1&quot;</span>, fls)

sce_unst_list &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (don <span class="cf">in</span> donors) {
    sce_unst_list[[don]] &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data/sces&quot;</span>, 
        <span class="kw">paste0</span>(<span class="st">&quot;sce_&quot;</span>, don, <span class="st">&quot;_with_clone_assignments.&quot;</span>, params<span class="op">$</span>callset, <span class="st">&quot;.rds&quot;</span>)))
    <span class="kw">cat</span>(<span class="kw">paste</span>(<span class="st">&quot;reading&quot;</span>, don, <span class="st">&quot;:   &quot;</span>, <span class="kw">ncol</span>(sce_unst_list[[don]]), <span class="st">&quot;cells.</span><span class="ch">\n</span><span class="st">&quot;</span>))
}</code></pre></div>
<pre><code>reading euts :    79 cells.
reading fawm :    53 cells.
reading feec :    75 cells.
reading fikt :    39 cells.
reading garx :    70 cells.
reading gesg :    105 cells.
reading heja :    50 cells.
reading hipn :    62 cells.
reading ieki :    58 cells.
reading joxm :    79 cells.
reading kuco :    48 cells.
reading laey :    55 cells.
reading lexy :    63 cells.
reading naju :    44 cells.
reading nusw :    60 cells.
reading oaaz :    38 cells.
reading oilg :    90 cells.
reading pipw :    107 cells.
reading puie :    41 cells.
reading qayj :    97 cells.
reading qolg :    36 cells.
reading qonc :    58 cells.
reading rozh :    91 cells.
reading sehl :    30 cells.
reading ualf :    89 cells.
reading vass :    37 cells.
reading vils :    37 cells.
reading vuna :    71 cells.
reading wahn :    82 cells.
reading wetu :    77 cells.
reading xugn :    35 cells.
reading zoxy :    88 cells.</code></pre>
<p>The starting point for differential expression analysis was a set of 32 donors, of which 31 donors had enough cells assigned to clones to conduct DE testing.</p>
<p>Summarise cell assignment information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">assignments_lst &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (don <span class="cf">in</span> donors) {
    assignments_lst[[don]] &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(
        <span class="kw">colData</span>(sce_unst_list[[don]])[, 
                                      <span class="kw">c</span>(<span class="st">&quot;donor_short_id&quot;</span>, <span class="st">&quot;highest_prob&quot;</span>, 
                                        <span class="st">&quot;assigned&quot;</span>, <span class="st">&quot;total_features&quot;</span>,
                                        <span class="st">&quot;total_counts_endogenous&quot;</span>, <span class="st">&quot;num_processed&quot;</span>)])
}
assignments &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, assignments_lst)</code></pre></div>
<p>85% of cells from these donors are assigned with confidence to a clone.</p>
<p>Load donor info including evidence for selection dynamics in donors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_donor_info &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/donor_info_070818.txt&quot;</span>)</code></pre></div>
</div>
<div id="genewise-de-results" class="section level2">
<h2>Genewise DE results</h2>
<p>We first look at differential expression at the level of individual genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fdr_thresh &lt;-<span class="st"> </span><span class="dv">1</span>
df_de_all_unst &lt;-<span class="st"> </span><span class="kw">data_frame</span>()
<span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]])) {
    tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table
    tmp<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table)
    ihw_res &lt;-<span class="st"> </span><span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">data =</span> tmp, <span class="dt">alpha =</span> <span class="fl">0.05</span>)
    tmp<span class="op">$</span>FDR &lt;-<span class="st"> </span><span class="kw">adj_pvalues</span>(ihw_res)
    tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]
    <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {
        tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor
        df_de_all_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_de_all_unst, tmp)
    }
}

df_ncells_de &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>, 
                              donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_cells =</span> <span class="kw">n</span>())
<span class="kw">colnames</span>(df_ncells_de)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span>

fdr_thresh &lt;-<span class="st"> </span><span class="fl">0.1</span>
df_de_sig_unst &lt;-<span class="st"> </span><span class="kw">data_frame</span>()
<span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]])) {
    tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table
    tmp<span class="op">$</span>gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(de_res[[<span class="st">&quot;qlf_list&quot;</span>]][[donor]]<span class="op">$</span>table)
    ihw_res &lt;-<span class="st"> </span><span class="kw">ihw</span>(PValue <span class="op">~</span><span class="st"> </span>logCPM, <span class="dt">data =</span> tmp, <span class="dt">alpha =</span> <span class="fl">0.05</span>)
    tmp<span class="op">$</span>FDR &lt;-<span class="st"> </span><span class="kw">adj_pvalues</span>(ihw_res)
    tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;</span><span class="st"> </span>fdr_thresh,]
    <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {
        tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor
        df_de_sig_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_de_sig_unst, tmp)
    }
}

df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> count)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">xend =</span> n_donors, <span class="dt">y =</span> count, <span class="dt">yend =</span> <span class="fl">0.1</span>), 
                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_y_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2000</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">20</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of donors in which significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;edgeR QL F test DE results&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/genewise-de-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
       
p1 &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> count)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">xend =</span> n_donors, <span class="dt">y =</span> count, <span class="dt">yend =</span> <span class="fl">0.1</span>), 
                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span>
<span class="st">    </span><span class="co">#coord_cartesian(ylim = c(1, 2200)) +</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dv">16</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of donors significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)
p1 </code></pre></div>
<p><img src="figure/differential_expression.Rmd/genewise-de-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_linscale.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_linscale.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_linscale.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p2 &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(gene, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">gsub</span>(<span class="st">&quot;ENSG.*_&quot;</span>, <span class="st">&quot;&quot;</span>, gene)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n_donors <span class="op">&gt;</span><span class="st"> </span><span class="fl">7.5</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(gene, n_donors, max))) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="dv">7</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span>
<span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of donors significant&quot;</span>)
p2</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_topgenes.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_topgenes.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_topgenes.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)


<span class="co">#cowplot::plot_grid(p1, p2, rel_heights = c(0.4, 0.6))</span>


df_donor_n_de &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>())
no_de_donor &lt;-<span class="st"> </span><span class="kw">unique</span>(df_de_all_unst[[<span class="st">&quot;donor&quot;</span>]])[<span class="op">!</span>(<span class="kw">unique</span>(df_de_all_unst[[<span class="st">&quot;donor&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span>df_donor_n_de[[<span class="st">&quot;donor&quot;</span>]])]
df_donor_n_de &lt;-<span class="st"> </span><span class="kw">rbind</span>(df_donor_n_de, <span class="kw">data_frame</span>(<span class="dt">donor =</span> no_de_donor, <span class="dt">count =</span> <span class="dv">0</span>))</code></pre></div>
<p>Permute gene labels to get a null distribution.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_de_nsig &lt;-<span class="st"> </span>df_de_all_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>())
df_nsig_ncells_de &lt;-<span class="st"> </span><span class="kw">full_join</span>(df_ncells_de, df_de_nsig)
df_nsig_ncells_de<span class="op">$</span>n_sig[<span class="kw">is.na</span>(df_nsig_ncells_de<span class="op">$</span>n_sig)] &lt;-<span class="st"> </span><span class="dv">0</span>

permute_gene_labels &lt;-<span class="st"> </span><span class="cf">function</span>(gene_names, n_de)  {
    sampled_genes &lt;-<span class="st"> </span><span class="kw">c</span>()
    <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(n_de))
        sampled_genes &lt;-<span class="st"> </span><span class="kw">c</span>(sampled_genes, <span class="kw">sample</span>(gene_names, <span class="dt">size =</span> n_de[i]))
    tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">table</span>(sampled_genes))
    df &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">n_donors =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">11</span>, <span class="dt">n_genes =</span> <span class="dv">0</span>)
    df[<span class="kw">names</span>(tab), <span class="dv">2</span>] &lt;-<span class="st"> </span>tab
    df
}


n_perm &lt;-<span class="st"> </span><span class="dv">1000</span>
df_perm &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(n_perm))
    df_perm[[i]] &lt;-<span class="st"> </span><span class="kw">permute_gene_labels</span>(<span class="kw">rownames</span>(de_res<span class="op">$</span>qlf_list<span class="op">$</span>vass<span class="op">$</span>table), 
                                        df_nsig_ncells_de[[<span class="st">&quot;n_sig&quot;</span>]])
df_perm &lt;-<span class="st"> </span><span class="kw">do.call</span>(<span class="st">&quot;rbind&quot;</span>, df_perm)
df_perm &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(df_perm, <span class="dt">data_type =</span> <span class="st">&quot;permuted&quot;</span>)

df_perm <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">min =</span> <span class="kw">min</span>(n_genes), 
                                             <span class="dt">median =</span> <span class="kw">median</span>(n_genes),
                                             <span class="dt">mean =</span> <span class="kw">mean</span>(n_genes), 
                                             <span class="dt">max =</span> <span class="kw">max</span>(n_genes))</code></pre></div>
<pre><code># A tibble: 11 x 5
   n_donors   min median     mean   max
      &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;
 1        1  3946   4120 4121.     4292
 2        2  2356   2466 2468.     2578
 3        3   821    897  896.      976
 4        4   177    223  223.      269
 5        5    23     40   40.3      59
 6        6     0      5    5.63     14
 7        7     0      0    0.595     4
 8        8     0      0    0.062     2
 9        9     0      0    0.007     1
10       10     0      0    0         0
11       11     0      0    0         0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ppp &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_genes =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">data_type =</span> <span class="st">&quot;observed&quot;</span>) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> n_genes)) <span class="op">+</span>
<span class="st">    </span><span class="co"># geom_segment(aes(x = n_donors, xend = n_donors, y = count, yend = 0.1), </span>
<span class="st">    </span><span class="co">#              colour = &quot;gray50&quot;) +</span>
<span class="co">#    geom_hline(yintercept = 0, linetype = 2) +</span>
<span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">group =</span> n_donors, <span class="dt">y =</span> n_genes, <span class="dt">colour =</span> data_type), 
                 <span class="dt">fill =</span> <span class="st">&quot;gray80&quot;</span>, <span class="dt">data =</span> df_perm, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour =</span> data_type), <span class="dt">shape =</span> <span class="dv">17</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),
                 <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),
                 <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">name =</span> <span class="st">&#39;&#39;</span>, 
                        <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;permuted&quot;</span> =<span class="st"> &quot;gray50&quot;</span>), 
                        <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span>, <span class="st">&quot;permuted&quot;</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">18</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.88</span>),
          <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">shape =</span> <span class="kw">c</span>(<span class="dv">17</span>, <span class="dv">19</span>))),
           <span class="dt">fill =</span> <span class="ot">FALSE</span>, <span class="dt">boxplot =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of donors significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)

<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>, <span class="dt">plot =</span> ppp)

df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(n_donors) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_genes =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">data_type =</span> <span class="st">&quot;observed&quot;</span>) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_donors, <span class="dt">y =</span> n_genes)) <span class="op">+</span>
<span class="st">    </span><span class="co"># geom_segment(aes(x = n_donors, xend = n_donors, y = count, yend = 0.1), </span>
<span class="st">    </span><span class="co">#              colour = &quot;gray50&quot;) +</span>
<span class="co">#    geom_hline(yintercept = 0, linetype = 2) +</span>
<span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">group =</span> n_donors, <span class="dt">y =</span> n_genes, <span class="dt">colour =</span> data_type), 
                 <span class="dt">fill =</span> <span class="st">&quot;gray80&quot;</span>, <span class="dt">data =</span> df_perm, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour =</span> data_type), <span class="dt">shape =</span> <span class="dv">17</span>, <span class="dt">size =</span> <span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">11</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),
                 <span class="dt">labels =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">2000</span>, <span class="dv">3000</span>),
                 <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">name =</span> <span class="st">&#39;&#39;</span>, 
                        <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;permuted&quot;</span> =<span class="st"> &quot;gray50&quot;</span>), 
                        <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;observed&quot;</span>, <span class="st">&quot;permuted&quot;</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">4500</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.88</span>),
          <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>()) <span class="op">+</span>
<span class="st">    </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">shape =</span> <span class="kw">c</span>(<span class="dv">17</span>, <span class="dv">19</span>))),
           <span class="dt">fill =</span> <span class="ot">FALSE</span>, <span class="dt">boxplot =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Number of donors significant (FDR &lt; 10%)&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/permute-de-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_de_n_sig_donors_n_sig_genes_sqrtscale_perm_skinny.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)</code></pre></div>
<p>Look at recurrently DE genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_gene_n_de &lt;-<span class="st"> </span>df_de_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, gene)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(gene) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(count)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">ensembl_gene_id =</span> <span class="kw">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, gene),
                  <span class="dt">hgnc_symbol =</span> <span class="kw">gsub</span>(<span class="st">&quot;.*_&quot;</span>, <span class="st">&quot;&quot;</span>, gene))
df_gene_n_de &lt;-<span class="st"> </span><span class="kw">left_join</span>(
    df_gene_n_de,
    dplyr<span class="op">::</span><span class="kw">select</span>(de_res<span class="op">$</span>qlf_pairwise<span class="op">$</span>joxm<span class="op">$</span>clone2_clone1<span class="op">$</span>table, 
                  ensembl_gene_id, hgnc_symbol, entrezid)
    )
df_gene_n_de &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(
    df_gene_n_de,
    <span class="dt">cell_cycle_growth =</span> (entrezid <span class="op">%in%</span><span class="st"> </span>
<span class="st">                             </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_G2M_CHECKPOINT,
                               Hs.H<span class="op">$</span>HALLMARK_MITOTIC_SPINDLE,
                               Hs.H<span class="op">$</span>HALLMARK_E2F_TARGETS)),
    <span class="dt">myc =</span> (entrezid <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_MYC_TARGETS_V1,
                           Hs.H<span class="op">$</span>HALLMARK_MYC_TARGETS_V2)),
    <span class="dt">emt =</span> (entrezid <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(Hs.H<span class="op">$</span>HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION))
)
df_gene_n_de <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(count <span class="op">&gt;=</span><span class="st"> </span><span class="dv">8</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw">datatable</span>(.)</code></pre></div>
<div id="htmlwidget-145d45e0d6a8b1c55880" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-145d45e0d6a8b1c55880">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62"],["ENSG00000146374_RSPO3","ENSG00000255526_NEDD8-MDP1","ENSG00000007944_MYLIP","ENSG00000101160_CTSZ","ENSG00000139835_GRTP1","ENSG00000144407_PTH2R","ENSG00000150394_CDH8","ENSG00000157399_ARSE","ENSG00000159753_RLTPR","ENSG00000168944_CEP120","ENSG00000214188_ST7-OT4","ENSG00000072041_SLC6A15","ENSG00000077157_PPP1R12B","ENSG00000106819_ASPN","ENSG00000123411_IKZF4","ENSG00000124610_HIST1H1A","ENSG00000125398_SOX9","ENSG00000127083_OMD","ENSG00000128573_FOXP2","ENSG00000146592_CREB5","ENSG00000146826_C7orf43","ENSG00000164124_TMEM144","ENSG00000213057_C1orf220","ENSG00000228169_PPIAP19","ENSG00000039139_DNAH5","ENSG00000060069_CTDP1","ENSG00000060140_STYK1","ENSG00000064763_FAR2","ENSG00000077943_ITGA8","ENSG00000095739_BAMBI","ENSG00000100784_RPS6KA5","ENSG00000105321_CCDC9","ENSG00000105852_PON3","ENSG00000106069_CHN2","ENSG00000112137_PHACTR1","ENSG00000113356_POLR3G","ENSG00000117245_KIF17","ENSG00000118292_C1orf54","ENSG00000141448_GATA6","ENSG00000143178_TBX19","ENSG00000143882_ATP6V1C2","ENSG00000156853_ZNF689","ENSG00000158402_CDC25C","ENSG00000164185_ZNF474","ENSG00000172346_CSDC2","ENSG00000172650_AGAP5","ENSG00000174137_FAM53A","ENSG00000178852_EFCAB13","ENSG00000180881_CAPS2","ENSG00000182325_FBXL6","ENSG00000184083_FAM120C","ENSG00000184381_PLA2G6","ENSG00000185666_SYN3","ENSG00000187800_PEAR1","ENSG00000197024_ZNF398","ENSG00000211448_DIO2","ENSG00000213889_PPM1N","ENSG00000214265_SNURF","ENSG00000232472_EEF1B2P3","ENSG00000235194_PPP1R3E","ENSG00000250959_GLUD1P3","ENSG00000251503_APITD1-CORT"],[11,11,10,10,10,10,10,10,10,10,10,9,9,9,9,9,9,9,9,9,9,9,9,9,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8],["ENSG00000146374","ENSG00000255526","ENSG00000007944","ENSG00000101160","ENSG00000139835","ENSG00000144407","ENSG00000150394","ENSG00000157399","ENSG00000159753","ENSG00000168944","ENSG00000214188","ENSG00000072041","ENSG00000077157","ENSG00000106819","ENSG00000123411","ENSG00000124610","ENSG00000125398","ENSG00000127083","ENSG00000128573","ENSG00000146592","ENSG00000146826","ENSG00000164124","ENSG00000213057","ENSG00000228169","ENSG00000039139","ENSG00000060069","ENSG00000060140","ENSG00000064763","ENSG00000077943","ENSG00000095739","ENSG00000100784","ENSG00000105321","ENSG00000105852","ENSG00000106069","ENSG00000112137","ENSG00000113356","ENSG00000117245","ENSG00000118292","ENSG00000141448","ENSG00000143178","ENSG00000143882","ENSG00000156853","ENSG00000158402","ENSG00000164185","ENSG00000172346","ENSG00000172650","ENSG00000174137","ENSG00000178852","ENSG00000180881","ENSG00000182325","ENSG00000184083","ENSG00000184381","ENSG00000185666","ENSG00000187800","ENSG00000197024","ENSG00000211448","ENSG00000213889","ENSG00000214265","ENSG00000232472","ENSG00000235194","ENSG00000250959","ENSG00000251503"],["RSPO3","NEDD8-MDP1","MYLIP","CTSZ","GRTP1","PTH2R","CDH8","ARSE","RLTPR","CEP120","ST7-OT4","SLC6A15","PPP1R12B","ASPN","IKZF4","HIST1H1A","SOX9","OMD","FOXP2","CREB5","C7orf43","TMEM144","C1orf220","PPIAP19","DNAH5","CTDP1","STYK1","FAR2","ITGA8","BAMBI","RPS6KA5","CCDC9","PON3","CHN2","PHACTR1","POLR3G","KIF17","C1orf54","GATA6","TBX19","ATP6V1C2","ZNF689","CDC25C","ZNF474","CSDC2","AGAP5","FAM53A","EFCAB13","CAPS2","FBXL6","FAM120C","PLA2G6","SYN3","PEAR1","ZNF398","DIO2","PPM1N","SNURF","EEF1B2P3","PPP1R3E","GLUD1P3","APITD1-CORT"],["84870","100528064","29116","1522","79774","5746","1006","415","146206","153241","338069","55117","4660","54829","64375","3024","6662","4958","93986","9586","55262","55314","400798",null,"1767","9150","55359","55711","8516","25805","9252","26093","5446","1124","221692","10622","57576","79630","2627","9095","245973","115509","995","133923","27254","729092","152877","124989","84698","26233","54954","8398","8224","375033","57541","1734","147699",null,null,"90673","2749","100526739"],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene<\/th>\n      <th>count<\/th>\n      <th>ensembl_gene_id<\/th>\n      <th>hgnc_symbol<\/th>\n      <th>entrezid<\/th>\n      <th>cell_cycle_growth<\/th>\n      <th>myc<\/th>\n      <th>emt<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<hr />
</div>
<div id="camera-results" class="section level2">
<h2>Camera results</h2>
<p>First, aggregate gene set enrichment results across all donors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fdr_thresh &lt;-<span class="st"> </span><span class="dv">1</span>
df_camera_all_unst &lt;-<span class="st"> </span><span class="kw">data_frame</span>()
<span class="cf">for</span> (geneset <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]])) {
    <span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]])) {
        <span class="cf">for</span> (coeff <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]])) {
            <span class="cf">for</span> (stat <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]])) {
                tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]][[stat]]
                tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]
                <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {
                    tmp[[<span class="st">&quot;collection&quot;</span>]] &lt;-<span class="st"> </span>geneset
                    tmp[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(tmp)
                    tmp[[<span class="st">&quot;coeff&quot;</span>]] &lt;-<span class="st"> </span>coeff
                    tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor
                    tmp[[<span class="st">&quot;stat&quot;</span>]] &lt;-<span class="st"> </span>stat
                    df_camera_all_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_camera_all_unst, tmp)
                }
            }
        }
    }
}

fdr_thresh &lt;-<span class="st"> </span><span class="fl">0.05</span>
df_camera_sig_unst &lt;-<span class="st"> </span><span class="kw">data_frame</span>()
<span class="cf">for</span> (geneset <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]])) {
    <span class="cf">for</span> (donor <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]])) {
        <span class="cf">for</span> (coeff <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]])) {
            <span class="cf">for</span> (stat <span class="cf">in</span> <span class="kw">names</span>(de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]])) {
                tmp &lt;-<span class="st"> </span>de_res[[<span class="st">&quot;camera&quot;</span>]][[geneset]][[donor]][[coeff]][[stat]]
                tmp &lt;-<span class="st"> </span>tmp[tmp<span class="op">$</span>FDR <span class="op">&lt;=</span><span class="st"> </span>fdr_thresh,]
                <span class="cf">if</span> (<span class="kw">nrow</span>(tmp) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) {
                    tmp[[<span class="st">&quot;collection&quot;</span>]] &lt;-<span class="st"> </span>geneset
                    tmp[[<span class="st">&quot;geneset&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">rownames</span>(tmp)
                    tmp[[<span class="st">&quot;coeff&quot;</span>]] &lt;-<span class="st"> </span>coeff
                    tmp[[<span class="st">&quot;donor&quot;</span>]] &lt;-<span class="st"> </span>donor
                    tmp[[<span class="st">&quot;stat&quot;</span>]] &lt;-<span class="st"> </span>stat
                    df_camera_sig_unst &lt;-<span class="st"> </span><span class="kw">bind_rows</span>(df_camera_sig_unst, tmp)
                }
            }
        }
    }
}

df_camera_sig_unst &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(
    df_camera_sig_unst,
    <span class="dt">contrast =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; - &quot;</span>, coeff),
    <span class="dt">msigdb_collection =</span> plyr<span class="op">::</span><span class="kw">mapvalues</span>(collection, <span class="dt">from =</span> <span class="kw">c</span>(<span class="st">&quot;c2&quot;</span>, <span class="st">&quot;c6&quot;</span>, <span class="st">&quot;H&quot;</span>), <span class="dt">to =</span> <span class="kw">c</span>(<span class="st">&quot;MSigDB curated (c2)&quot;</span>, <span class="st">&quot;MSigDB oncogenic (c6)&quot;</span>, <span class="st">&quot;MSigDB Hallmark&quot;</span>)))

df_camera_all_unst &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(
    df_camera_all_unst,
    <span class="dt">contrast =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; - &quot;</span>, coeff),
    <span class="dt">msigdb_collection =</span> plyr<span class="op">::</span><span class="kw">mapvalues</span>(collection, <span class="dt">from =</span> <span class="kw">c</span>(<span class="st">&quot;c2&quot;</span>, <span class="st">&quot;c6&quot;</span>, <span class="st">&quot;H&quot;</span>), <span class="dt">to =</span> <span class="kw">c</span>(<span class="st">&quot;MSigDB curated (c2)&quot;</span>, <span class="st">&quot;MSigDB oncogenic (c6)&quot;</span>, <span class="st">&quot;MSigDB Hallmark&quot;</span>)))</code></pre></div>
<p>We now have a dataframe for significant (FDR &lt;5%) results from the camera gene set enrichment results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(df_camera_sig_unst)</code></pre></div>
<pre><code># A tibble: 6 x 11
  NGenes Direction   PValue      FDR collection geneset coeff donor stat 
   &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
1      6 Down      6.43e-15 3.02e-11 c2         VALK_A… clon… euts  signF
2      7 Down      2.29e-11 5.37e- 8 c2         WANG_R… clon… euts  signF
3      5 Down      7.80e-11 1.22e- 7 c2         GUTIER… clon… euts  signF
4      8 Down      1.35e- 9 1.58e- 6 c2         KUMAMO… clon… euts  signF
5     11 Down      4.95e- 9 4.65e- 6 c2         VALK_A… clon… euts  signF
6      4 Down      2.86e- 8 1.92e- 5 c2         REACTO… clon… euts  signF
# ... with 2 more variables: contrast &lt;chr&gt;, msigdb_collection &lt;chr&gt;</code></pre>
<p>And a dataframe with all results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(df_camera_all_unst)</code></pre></div>
<pre><code># A tibble: 6 x 11
  NGenes Direction  PValue   FDR collection geneset coeff donor stat 
   &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;
1      7 Down      2.88e-4 0.771 c2         WANG_R… clon… euts  logFC
2     10 Down      4.73e-4 0.771 c2         BIOCAR… clon… euts  logFC
3     15 Up        8.30e-4 0.771 c2         LI_CIS… clon… euts  logFC
4     14 Down      1.07e-3 0.771 c2         MOROSE… clon… euts  logFC
5      5 Down      1.12e-3 0.771 c2         IYENGA… clon… euts  logFC
6     15 Up        1.19e-3 0.771 c2         PID_TC… clon… euts  logFC
# ... with 2 more variables: contrast &lt;chr&gt;, msigdb_collection &lt;chr&gt;</code></pre>
<p>For now, focus on gene set enrichment results computed using log-fold change statistics for pairwise comparisons of clones estimated from the edgeR QL-F models.</p>
<p>We can look at all significant results summarised by donor, geneset and pairwise contrast of clones.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">x =</span> donor, <span class="dt">colour =</span> contrast)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_sina</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_grid</span>(contrast <span class="op">~</span><span class="st"> </span>msigdb_collection) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_colour_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<p><img src="figure/differential_expression.Rmd/camera-allcontr-allsets-sig-bydonor-pvals-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Similarly, we can look at all results summarised by donor, geneset and pairwise contrast of clones.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_camera_all_unst <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(FDR), <span class="dt">x =</span> donor, <span class="dt">colour =</span> contrast)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_sina</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="op">-</span><span class="kw">log10</span>(<span class="fl">0.05</span>), <span class="dt">linetype =</span> <span class="dv">2</span>, <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_grid</span>(contrast <span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_colour_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>()</code></pre></div>
<p><img src="figure/differential_expression.Rmd/camera-allcontr-allsets-all-bydonor-fdr-1.png" width="1056" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_all_results.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_all_results.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_all_results.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">16</span>, <span class="dt">width =</span> <span class="dv">14</span>)</code></pre></div>
<p>We can check the number of significant gene sets for each donor, for each MSigDB gene set collection.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">print</span>(<span class="dt">n =</span> <span class="ot">Inf</span>)</code></pre></div>
<pre><code># A tibble: 67 x 3
# Groups:   donor [?]
   donor msigdb_collection     n_sig
   &lt;chr&gt; &lt;chr&gt;                 &lt;int&gt;
 1 fawm  MSigDB curated (c2)     348
 2 fawm  MSigDB Hallmark           7
 3 fawm  MSigDB oncogenic (c6)     3
 4 feec  MSigDB curated (c2)     256
 5 feec  MSigDB Hallmark           4
 6 feec  MSigDB oncogenic (c6)     3
 7 fikt  MSigDB curated (c2)     426
 8 fikt  MSigDB Hallmark           9
 9 fikt  MSigDB oncogenic (c6)     4
10 garx  MSigDB curated (c2)     244
11 garx  MSigDB Hallmark           3
12 garx  MSigDB oncogenic (c6)     7
13 gesg  MSigDB curated (c2)       4
14 heja  MSigDB curated (c2)       7
15 hipn  MSigDB curated (c2)      99
16 hipn  MSigDB Hallmark           5
17 hipn  MSigDB oncogenic (c6)     1
18 ieki  MSigDB curated (c2)     480
19 ieki  MSigDB Hallmark           9
20 ieki  MSigDB oncogenic (c6)    10
21 joxm  MSigDB curated (c2)     150
22 joxm  MSigDB Hallmark           3
23 joxm  MSigDB oncogenic (c6)     1
24 kuco  MSigDB curated (c2)       1
25 laey  MSigDB curated (c2)     502
26 laey  MSigDB Hallmark           7
27 laey  MSigDB oncogenic (c6)    19
28 lexy  MSigDB curated (c2)     546
29 lexy  MSigDB Hallmark           8
30 lexy  MSigDB oncogenic (c6)     8
31 naju  MSigDB curated (c2)      99
32 naju  MSigDB Hallmark           2
33 naju  MSigDB oncogenic (c6)    10
34 oilg  MSigDB curated (c2)     102
35 oilg  MSigDB Hallmark           1
36 oilg  MSigDB oncogenic (c6)     1
37 pipw  MSigDB curated (c2)       1
38 puie  MSigDB curated (c2)       1
39 qayj  MSigDB curated (c2)     152
40 qayj  MSigDB Hallmark           5
41 qayj  MSigDB oncogenic (c6)     4
42 qolg  MSigDB curated (c2)     293
43 qolg  MSigDB Hallmark           3
44 qolg  MSigDB oncogenic (c6)     9
45 qonc  MSigDB curated (c2)     382
46 qonc  MSigDB Hallmark           5
47 qonc  MSigDB oncogenic (c6)     2
48 rozh  MSigDB curated (c2)     553
49 rozh  MSigDB Hallmark          10
50 rozh  MSigDB oncogenic (c6)    10
51 sehl  MSigDB curated (c2)      59
52 sehl  MSigDB Hallmark           2
53 sehl  MSigDB oncogenic (c6)     2
54 ualf  MSigDB curated (c2)     378
55 ualf  MSigDB Hallmark           3
56 ualf  MSigDB oncogenic (c6)     4
57 vass  MSigDB curated (c2)     270
58 vass  MSigDB Hallmark           8
59 vass  MSigDB oncogenic (c6)     3
60 wahn  MSigDB curated (c2)     129
61 wahn  MSigDB Hallmark           2
62 wetu  MSigDB curated (c2)       3
63 xugn  MSigDB curated (c2)     120
64 xugn  MSigDB Hallmark           4
65 zoxy  MSigDB curated (c2)     488
66 zoxy  MSigDB Hallmark          11
67 zoxy  MSigDB oncogenic (c6)    19</code></pre>
<p>We can look at the number of significant gene sets for each donor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## simpler version
df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">sum</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_sig, <span class="dt">x =</span> donor)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant genesets (FDR &lt; 5%)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB gene set enrichment by donor&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/alldonors_camera_enrichment_by_donor_simple-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</code></pre></div>
<p>We can look at the effect of the the number of cells for each donor on the DE results obtained.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncells_by_donor &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="ot">NA</span>, <span class="kw">length</span>(sce_unst_list))
<span class="kw">names</span>(ncells_by_donor) &lt;-<span class="st"> </span><span class="kw">names</span>(sce_unst_list)
<span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(sce_unst_list))
    ncells_by_donor[don] &lt;-<span class="st"> </span><span class="kw">ncol</span>(sce_unst_list[[don]])

df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor, msigdb_collection) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_sig =</span> <span class="kw">sum</span>(FDR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() -&gt;<span class="st"> </span>df_to_plot
df_to_plot &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df_to_plot, 
                        <span class="kw">data_frame</span>(<span class="dt">donor =</span> <span class="kw">names</span>(ncells_by_donor),
                                   <span class="dt">ncells =</span> ncells_by_donor))
df_to_plot <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor =</span> <span class="kw">factor</span>(donor, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(donor))))) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_sig, <span class="dt">x =</span> donor, <span class="dt">size =</span> ncells)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="st"> </span>msigdb_collection, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Accent&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Donor&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of significant genesets (FDR &lt; 5%)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB gene set enrichment by donor&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/alldonors_camera_enrichment_by_donor_simple_size_by_ncells-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_by_donor_simple_size_by_ncells.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="dv">10</span>)</code></pre></div>
<div id="hallmark-gene-set" class="section level3">
<h3>Hallmark gene set</h3>
<p>Focus now on looking at DE results for the MSigDB Hallmark gene set (50 of the best-characterised gene sets as determined by MSigDB).</p>
<p>Look at the gene sets that are found to be enriched in multiple donors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Hallmark geneset
df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max))) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span>
<span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_bw</span>(<span class="dv">14</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of donors significant&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/alldonors_camera_enrichment_H_by_geneset-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_H_by_geneset.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_H_by_geneset.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/alldonors_camera_enrichment_H_by_geneset.svg&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">7</span>, <span class="dt">width =</span> <span class="fl">9.5</span>)
## number of donors with at least one significant geneset
tmp &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(geneset)
<span class="kw">unique</span>(tmp[[<span class="st">&quot;donor&quot;</span>]])</code></pre></div>
<pre><code> [1] &quot;fawm&quot; &quot;feec&quot; &quot;fikt&quot; &quot;garx&quot; &quot;hipn&quot; &quot;ieki&quot; &quot;joxm&quot; &quot;laey&quot; &quot;lexy&quot; &quot;naju&quot;
[11] &quot;oilg&quot; &quot;qayj&quot; &quot;qolg&quot; &quot;qonc&quot; &quot;rozh&quot; &quot;sehl&quot; &quot;ualf&quot; &quot;vass&quot; &quot;wahn&quot; &quot;xugn&quot;
[21] &quot;zoxy&quot;</code></pre>
<p>21 donors have at least one significantly enriched Hallmark gene set.</p>
<p>For gene sets related directly to cell cycle and growth, we see contrasts being both up- and down- regulated, but for EMT, coagulation and angiogenesis pathways, we only see these down-regulated.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(geneset, Direction) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max),
             <span class="dt">colour =</span> Direction)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.7</span>, <span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dt">width =</span> <span class="fl">0.5</span>)) <span class="op">+</span>
<span class="st">  </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_bw</span>(<span class="dv">16</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of donors significant&quot;</span>)  <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Camera MSigDB Hallmark gene set enrichment&quot;</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/alldonors_camera_enrichment_H_by_geneset_by_dir-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap-of-results-for-camera-hallmark-geneset-testing" class="section level2">
<h2>Heatmap of results for camera Hallmark geneset testing</h2>
<p>We can get an overview of all the Hallmark gene set results by producing a heatmap, first showing just the significant (FDR &lt; 5%) results across all donors and pairwise contrasts of clones.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repeated_sig_H_genesets &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>, stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(n_donors) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n_donors <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) 
repeated_sig_H_genesets_vec &lt;-<span class="st"> </span><span class="kw">unique</span>(repeated_sig_H_genesets[[<span class="st">&quot;geneset&quot;</span>]])
repeated_sig_H_genesets_vec &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, 
                                                   repeated_sig_H_genesets_vec))
     
df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> 
                      <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast)) -&gt;
<span class="st">    </span>df_4_heatmap

div_lines &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;: c.*&quot;</span>, <span class="st">&quot;&quot;</span>,
     <span class="kw">sort</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(df_4_heatmap[[<span class="st">&quot;donor&quot;</span>]], <span class="st">&quot;: &quot;</span>, 
                        df_4_heatmap[[<span class="st">&quot;contrast&quot;</span>]])))) <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>cumsum <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span>
    
df_4_heatmap <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> geneset, <span class="dt">fill =</span> Direction)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> div_lines, <span class="dt">colour =</span> <span class="st">&quot;gray70&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightgoldenrod1&quot;</span>, <span class="st">&quot;sienna1&quot;</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/top_genesets_H_direction_heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap.png&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">6</span>, <span class="dt">width =</span> <span class="dv">12</span>)</code></pre></div>
<p>We can do the same for all results for the gene sets that are significantly enriched in at least two donors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> 
                      <span class="kw">factor</span>(geneset, <span class="dt">levels =</span> repeated_sig_H_genesets_vec)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, <span class="st">&quot;: &quot;</span>, contrast)) -&gt;
<span class="st">    </span>df_4_heatmap_all

df_4_heatmap_all &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(
    df_4_heatmap_all,
    <span class="dt">minlog10P =</span> <span class="kw">cut</span>(<span class="op">-</span><span class="kw">log10</span>(PValue), <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">30</span>)))

div_lines_all &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;: c.*&quot;</span>, <span class="st">&quot;&quot;</span>,
     <span class="kw">sort</span>(<span class="kw">unique</span>(<span class="kw">paste0</span>(df_4_heatmap_all[[<span class="st">&quot;donor&quot;</span>]], <span class="st">&quot;: &quot;</span>, 
                        df_4_heatmap_all[[<span class="st">&quot;contrast&quot;</span>]])))) <span class="op">%&gt;%</span><span class="st"> </span>table <span class="op">%&gt;%</span><span class="st"> </span>cumsum <span class="op">+</span><span class="st"> </span><span class="fl">0.5</span>

pp &lt;-<span class="st"> </span>df_4_heatmap_all <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> id, <span class="dt">y =</span> geneset, <span class="dt">fill =</span> Direction, <span class="dt">alpha =</span> minlog10P)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="dv">1</span>, <span class="dt">data =</span> df_4_heatmap, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> div_lines_all, <span class="dt">colour =</span> <span class="st">&quot;gray70&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;lightgoldenrod1&quot;</span>, <span class="st">&quot;sienna1&quot;</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_alpha_discrete</span>(<span class="dt">name =</span> <span class="st">&quot;-log10(P)&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Donor and clone comparison&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">60</span>, <span class="dt">hjust =</span> <span class="dv">1</span>),
          <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>)
pp    </code></pre></div>
<p><img src="figure/differential_expression.Rmd/top_genesets_H_direction_heatmap_all_contrasts-1.png" width="1728" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</code></pre></div>
<p>Finally, we can add a panel to this figure showing the number of donors in which each of these gene sets is significantly enriched.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Hallmark geneset
pp_nsig &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> <span class="kw">reorder</span>(geneset, n_donors, max))) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">xend =</span> <span class="kw">reorder</span>(geneset, n_donors, max), <span class="dt">yend =</span> <span class="dv">0</span>),
                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>, <span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of donors significant&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>())

prow &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(pp <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>), 
                  pp_nsig, <span class="dt">align =</span> <span class="st">&#39;h&#39;</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">7</span>, <span class="dv">1</span>))
lgnd &lt;-<span class="st"> </span><span class="kw">get_legend</span>(pp)
<span class="kw">plot_grid</span>(prow, lgnd, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">3</span>, .<span class="dv">3</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_direction_heatmap_all_contrasts_with_nsig_donors.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</code></pre></div>
</div>
<div id="correlation-of-gene-set-results-and-genes-contained" class="section level2">
<h2>Correlation of gene set results and genes contained</h2>
<p>Let’s look at the correlation between gene set results (Spearman correlation of signed -log10(P-values) from <em>camera</em> tests) and compare to the proportion of genes overlapping between pairs of gene sets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">repeated_sig_H_genesets_vec2 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, 
                                       <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, repeated_sig_H_genesets_vec))
## all results
df_H_pvals &lt;-<span class="st"> </span>df_camera_all_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>, geneset <span class="op">%in%</span><span class="st"> </span>repeated_sig_H_genesets_vec2) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">donor_coeff =</span> <span class="kw">paste</span>(donor, coeff, <span class="dt">sep =</span> <span class="st">&quot;.&quot;</span>),
                <span class="dt">sign =</span> <span class="kw">ifelse</span>(Direction <span class="op">==</span><span class="st"> &quot;Down&quot;</span>, <span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>),
                <span class="dt">signed_P =</span> sign <span class="op">*</span><span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(PValue)) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(geneset, donor_coeff, signed_P) <span class="op">%&gt;%</span>
<span class="st">  </span>tidyr<span class="op">::</span><span class="kw">spread</span>(<span class="dt">key =</span> donor_coeff, <span class="dt">value =</span> signed_P)

mat_H_pvals &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(df_H_pvals[, <span class="op">-</span><span class="dv">1</span>])
<span class="kw">rownames</span>(mat_H_pvals) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, df_H_pvals[[<span class="dv">1</span>]]))
cor_H_pvals &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">t</span>(mat_H_pvals), <span class="dt">method =</span> <span class="st">&quot;spearman&quot;</span>)
p.mat &lt;-<span class="st"> </span><span class="kw">cor_pmat</span>(<span class="kw">t</span>(mat_H_pvals))
<span class="kw">ggcorrplot</span>(cor_H_pvals, <span class="dt">hc.order =</span> <span class="ot">TRUE</span>, <span class="dt">p.mat =</span> p.mat, <span class="dt">insig =</span> <span class="st">&quot;blank&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),
        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())</code></pre></div>
<p><img src="figure/differential_expression.Rmd/corr-maps-1.png" width="1248" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hclust_cor &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">as.dist</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>cor_H_pvals))
corrplot1 &lt;-<span class="st"> </span><span class="kw">ggcorrplot</span>(cor_H_pvals[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order], 
           <span class="dt">p.mat =</span> p.mat[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order], <span class="dt">insig =</span> <span class="st">&quot;blank&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),
        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())

mat_H_gene_overlap &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow =</span> <span class="kw">nrow</span>(cor_H_pvals), <span class="dt">ncol =</span> <span class="kw">ncol</span>(cor_H_pvals),
                             <span class="dt">dimnames =</span> <span class="kw">dimnames</span>(cor_H_pvals))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(repeated_sig_H_genesets_vec2)) {
  <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq_along</span>(repeated_sig_H_genesets_vec2)) {
    gs1 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="kw">rownames</span>(mat_H_gene_overlap)[i]))
    gs2 &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="kw">rownames</span>(mat_H_gene_overlap)[j]))
    mat_H_gene_overlap[i, j] &lt;-<span class="st"> </span><span class="kw">mean</span>(Hs.H[[gs1]] <span class="op">%in%</span><span class="st"> </span>Hs.H[[gs2]])
  }
}

corrplot2 &lt;-<span class="st"> </span><span class="kw">ggcorrplot</span>(mat_H_gene_overlap[hclust_cor<span class="op">$</span>order, hclust_cor<span class="op">$</span>order]) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_gradient</span>(<span class="dt">name =</span> <span class="st">&quot;Gene set</span><span class="ch">\n</span><span class="st">overlap&quot;</span>, <span class="dt">low =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">high =</span> <span class="st">&quot;black&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(),
        <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>())
corrplot2</code></pre></div>
<p><img src="figure/differential_expression.Rmd/corr-maps-2.png" width="1248" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">corrplot3 &lt;-<span class="st"> </span>corrplot2 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>())</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_grid</span>(corrplot1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)), 
          corrplot3 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>)), 
          <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">axis =</span> <span class="st">&quot;b&quot;</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.58</span>, <span class="fl">0.42</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/corr-plot-combined-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_corrplots.png&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_corrplots.pdf&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_corrplots.svg&quot;</span>, <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">20</span>)</code></pre></div>
<p>Plot gene set correlation with the number of donors in which each gene set is significant.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pp_nsig &lt;-<span class="st"> </span>df_camera_sig_unst <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(collection <span class="op">==</span><span class="st"> &quot;H&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(stat <span class="op">==</span><span class="st"> &quot;logFC&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(geneset) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">id =</span> <span class="kw">paste0</span>(donor, geneset)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(id, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_donors =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(geneset, n_donors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, <span class="kw">gsub</span>(<span class="st">&quot;HALLMARK_&quot;</span>, <span class="st">&quot;&quot;</span>, geneset))) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">factor</span>(
      geneset, <span class="dt">levels =</span> <span class="kw">rownames</span>(mat_H_gene_overlap)[hclust_cor<span class="op">$</span>order])) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y =</span> n_donors, <span class="dt">x =</span> geneset)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">xend =</span> geneset, <span class="dt">yend =</span> <span class="dv">0</span>),
                 <span class="dt">colour =</span> <span class="st">&quot;gray50&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">4</span>, <span class="dt">colour =</span> <span class="st">&quot;gray30&quot;</span>, <span class="dt">alpha =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span>ggthemes<span class="op">::</span><span class="kw">scale_colour_tableau</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Gene set&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Number of donors</span><span class="ch">\n</span><span class="st">significant&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.y =</span> <span class="kw">element_blank</span>(),
          <span class="dt">axis.line.y =</span> <span class="kw">element_blank</span>())

<span class="kw">ggdraw</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">draw_plot</span>(corrplot1 <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>), 
            <span class="dt">x =</span> <span class="dv">0</span>,  <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.8</span>, <span class="dt">scale =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">draw_plot</span>(pp_nsig, 
            <span class="dt">x =</span> <span class="fl">0.785</span>,  <span class="dt">y =</span> <span class="fl">0.213</span>, <span class="dt">width =</span> <span class="fl">0.2</span>, <span class="dt">height =</span> <span class="fl">0.685</span>)</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_corrplot_with_nsig_donor.png&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">11</span>)
<span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/top_genesets_H_corrplot_with_nsig_donor.pdf&quot;</span>, 
       <span class="dt">height =</span> <span class="dv">9</span>, <span class="dt">width =</span> <span class="dv">11</span>)</code></pre></div>
</div>
<div id="linking-de-to-selection" class="section level2">
<h2>Linking DE to selection</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_donor_info &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/donor_info_070818.txt&quot;</span>)
df_donor_info &lt;-<span class="st"> </span><span class="kw">as_data_frame</span>(df_donor_info)
df_donor_info<span class="op">$</span>donor &lt;-<span class="st"> </span>df_donor_info<span class="op">$</span>donor_short

df_ncells_de &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>, 
                              donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">n_cells =</span> <span class="kw">n</span>())
<span class="kw">colnames</span>(df_ncells_de)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span>

df_prop_assigned &lt;-<span class="st"> </span>assignments <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(donor_short_id <span class="op">%in%</span><span class="st"> </span><span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(donor_short_id) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise</span>(<span class="dt">prop_assigned =</span> <span class="kw">mean</span>(assigned <span class="op">!=</span><span class="st"> &quot;unassigned&quot;</span>))
<span class="kw">colnames</span>(df_prop_assigned)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;donor&quot;</span>

df_nvars_by_cat &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;output/nvars_by_category_by_donor.tsv&quot;</span>)
df_nvars_by_cat_wd &lt;-<span class="st"> </span>tidyr<span class="op">::</span><span class="kw">spread</span>(
  df_nvars_by_cat[, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], consequence, n_vars_all_genes)
df_nvars_by_cat_wd &lt;-<span class="st"> </span><span class="kw">left_join</span>(
  <span class="kw">summarise</span>(<span class="kw">group_by</span>(df_nvars_by_cat, donor), <span class="dt">nvars_all =</span> <span class="kw">sum</span>(n_vars_all_genes)),
  df_nvars_by_cat_wd
)
df_nvars_by_cat_wd &lt;-<span class="st"> </span>df_nvars_by_cat <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(consequence <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;missense&quot;</span>, <span class="st">&quot;splicing&quot;</span>, <span class="st">&quot;nonsense&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">nvars_misnonspli =</span> <span class="kw">sum</span>(n_vars_all_genes)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">left_join</span>(., df_nvars_by_cat_wd)

df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_ncells_de, df_donor_info)
df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_prop_assigned, df_donor_info)
df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_donor_n_de, df_donor_info)
df_donor_info<span class="op">$</span>n_de_genes &lt;-<span class="st"> </span>df_donor_info<span class="op">$</span>count
df_donor_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(df_donor_info, df_nvars_by_cat_wd)

nbglm_nde &lt;-<span class="st"> </span>MASS<span class="op">::</span><span class="kw">glm.nb</span>(n_de_genes <span class="op">~</span><span class="st"> </span>n_cells, <span class="dt">data =</span> df_donor_info)
df_nbglm_nde &lt;-<span class="st"> </span>broom<span class="op">::</span><span class="kw">augment</span>(nbglm_nde) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">left_join</span>(df_donor_info)

## n_de vs n_cells
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> n_cells, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of cells&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_genes_vs_n_cells.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)

## selection, n_de resid boxplot
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> .resid)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span>
<span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_boxplot.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">4.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

<span class="kw">summary</span>(<span class="kw">lm</span>(.resid <span class="op">~</span><span class="st"> </span>selection, <span class="dt">data =</span> df_nbglm_nde))</code></pre></div>
<pre><code>
Call:
lm(formula = .resid ~ selection, data = df_nbglm_nde)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.7896 -0.4823 -0.0568  0.4630  3.3122 

Coefficients:
                      Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)            -0.9100     0.4242  -2.145   0.0408 *
selectionselected       0.7766     0.5612   1.384   0.1773  
selectionundetermined   0.5391     0.4934   1.093   0.2839  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.039 on 28 degrees of freedom
Multiple R-squared:  0.06604,   Adjusted R-squared:  -0.0006725 
F-statistic: 0.9899 on 2 and 28 DF,  p-value: 0.3842</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## selection, n_de (sqrt scale) boxplot
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> selection, <span class="dt">y =</span> n_de_genes)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_violin</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">alpha =</span> <span class="fl">0.7</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.alpha =</span> <span class="dv">0</span>, <span class="dt">width =</span> <span class="fl">0.2</span>) <span class="op">+</span>
<span class="st">  </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_quasirandom</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> selection), <span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Inferred selection status&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_flip</span>()</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_sqrt_selection_boxplot.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (resids) vs goodness of fit cumul. mutation model
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_vs_gof_cumul_mut_model.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)


## n_de (sqrt scale) vs goodness of fit cumul. mutation model
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_ntrtestr, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: cumulative mutations&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_sqrt_selection_vs_gof_cumul_mut_model.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (resids) vs goodness of fit NB model
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> rsq_negbinfit, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Goodness of fit: negative binomial distribution&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_vs_gof_negbin_model.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (resids) vs mutational load (all)
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-7.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_vs_n_somatic_vars_all.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (sqrt scale) vs mutational load (all)
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_all, <span class="dt">y =</span> n_de_genes, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic variants&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_y_sqrt</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">500</span>, <span class="dv">1000</span>, <span class="dv">1500</span>, <span class="dv">2000</span>, <span class="dv">2500</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-8.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_sqrt_selection_vs_n_somatic_vars_all.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (resids) vs mutational load (missense)
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> missense, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of somatic missense variants&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-9.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_vs_n_somatic_vars_missense.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">5.5</span>, <span class="dt">width =</span> <span class="fl">6.5</span>)

## n_de (resids) vs mutational load (missense, nonsense, splicing)
df_nbglm_nde <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">selection =</span> <span class="kw">factor</span>(
    selection, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;undetermined&quot;</span>, <span class="st">&quot;selected&quot;</span>))) <span class="op">%&gt;%</span>
<span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> nvars_misnonspli, <span class="dt">y =</span> .resid, <span class="dt">fill =</span> selection)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="kw">aes</span>(<span class="dt">group =</span> <span class="dv">1</span>), <span class="dt">colour =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">level =</span> <span class="fl">0.9</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">3</span>, <span class="dt">shape =</span> <span class="dv">21</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of DE genes (residual from NB GLM)&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Number of missense, nonsense &amp; splicing variants&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;dodgerblue&quot;</span>, <span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_x_log10</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">500</span>))</code></pre></div>
<p><img src="figure/differential_expression.Rmd/unnamed-chunk-5-10.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggsave</span>(<span class="st">&quot;figures/differential_expression/n_de_resid_selection_vs_n_somatic_vars_misnonspli.png&quot;</span>, 
       <span class="dt">height =</span> <span class="fl">6.5</span>, <span class="dt">width =</span> <span class="fl">5.5</span>)

<span class="co"># </span>
<span class="co"># fdr_thresh &lt;- 0.1</span>
<span class="co"># df_de_sig_unst &lt;- data_frame()</span>
<span class="co"># for (donor in names(de_res[[&quot;qlf_list&quot;]])) {</span>
<span class="co">#     tmp &lt;- de_res[[&quot;qlf_list&quot;]][[donor]]$table</span>
<span class="co">#     tmp$gene &lt;- rownames(de_res[[&quot;qlf_list&quot;]][[donor]]$table)</span>
<span class="co">#     ihw_res &lt;- ihw(PValue ~ logCPM, data = tmp, alpha = 0.05)</span>
<span class="co">#     tmp$FDR &lt;- adj_pvalues(ihw_res)</span>
<span class="co">#     tmp &lt;- tmp[tmp$FDR &lt; fdr_thresh,]</span>
<span class="co">#     if (nrow(tmp) &gt; 0.5) {</span>
<span class="co">#         tmp[[&quot;donor&quot;]] &lt;- donor</span>
<span class="co">#         df_de_sig_unst &lt;- bind_rows(df_de_sig_unst, tmp)</span>
<span class="co">#     }</span>
<span class="co"># }</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># df_donor_n_de &lt;- df_de_sig_unst %&gt;% </span>
<span class="co">#     group_by(gene) %&gt;% </span>
<span class="co">#     dplyr::mutate(id = paste0(donor, gene)) %&gt;% distinct(id, .keep_all = TRUE) %&gt;%</span>
<span class="co">#     group_by(donor) %&gt;%</span>
<span class="co">#     summarise(count = n())</span></code></pre></div>
</div>
<div id="linking-pathway-results-to-selection" class="section level2">
<h2>Linking pathway results to selection</h2>
<p>Not yet implemented</p>
</div>
<div id="compare-pathway-results-to-clone-prevalence" class="section level2">
<h2>Compare pathway results to clone prevalence</h2>
<p>Hallmark gene sets. Not yet implemented.</p>
</div>
<div id="write-de-and-pathway-results-to-file" class="section level2">
<h2>Write DE and pathway results to file</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Write DE results to file:
<span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>qlf_list)) {
  de_res<span class="op">$</span>qlf_list[[don]]<span class="op">$</span>table <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">rownames</span>(.), <span class="dt">FDR =</span> IHW<span class="op">::</span><span class="kw">adj_pvalues</span>(IHW<span class="op">::</span><span class="kw">ihw</span>(PValue, logCPM, <span class="dt">alpha =</span> <span class="fl">0.1</span>))) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(
      <span class="kw">paste0</span>(<span class="st">&quot;output/differential_expression/&quot;</span>, don, <span class="st">&quot;_qlf_de_results.tsv&quot;</span>))
}

<span class="cf">for</span> (don <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H)) {
  <span class="cf">for</span> (cntrst <span class="cf">in</span> <span class="kw">names</span>(de_res<span class="op">$</span>camera<span class="op">$</span>H[[don]])) {
    de_res<span class="op">$</span>camera<span class="op">$</span>H[[don]][[cntrst]]<span class="op">$</span>logFC <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneset =</span> <span class="kw">rownames</span>(.)) <span class="op">%&gt;%</span>
<span class="st">      </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(
        <span class="kw">paste0</span>(<span class="st">&quot;output/differential_expression/&quot;</span>, don, <span class="st">&quot;_camera_hallmark_geneset_results_&quot;</span>, cntrst, <span class="st">&quot;.tsv&quot;</span>))    
  }
}</code></pre></div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw">session_info</span>()</code></pre></div>
<pre><code> setting  value                       
 version  R version 3.5.1 (2018-07-02)
 system   x86_64, darwin15.6.0        
 ui       X11                         
 language (EN)                        
 collate  en_GB.UTF-8                 
 tz       Europe/London               
 date     2018-08-17                  

 package              * version   date       source                
 AnnotationDbi        * 1.42.1    2018-05-08 Bioconductor          
 assertthat             0.2.0     2017-04-11 CRAN (R 3.5.0)        
 backports              1.1.2     2017-12-13 CRAN (R 3.5.0)        
 base                 * 3.5.1     2018-07-05 local                 
 beeswarm               0.2.3     2016-04-25 CRAN (R 3.5.0)        
 bindr                  0.1.1     2018-03-13 CRAN (R 3.5.0)        
 bindrcpp             * 0.2.2     2018-03-29 CRAN (R 3.5.0)        
 Biobase              * 2.40.0    2018-05-01 Bioconductor          
 BiocGenerics         * 0.26.0    2018-05-01 Bioconductor          
 BiocParallel         * 1.14.2    2018-07-08 Bioconductor          
 bit                    1.1-14    2018-05-29 CRAN (R 3.5.0)        
 bit64                  0.9-7     2017-05-08 CRAN (R 3.5.0)        
 bitops                 1.0-6     2013-08-17 CRAN (R 3.5.0)        
 blob                   1.1.1     2018-03-25 CRAN (R 3.5.0)        
 broom                * 0.5.0     2018-07-17 CRAN (R 3.5.0)        
 cellranger             1.1.0     2016-07-27 CRAN (R 3.5.0)        
 cli                    1.0.0     2017-11-05 CRAN (R 3.5.0)        
 colorspace             1.3-2     2016-12-14 CRAN (R 3.5.0)        
 compiler               3.5.1     2018-07-05 local                 
 cowplot              * 0.9.3     2018-07-15 CRAN (R 3.5.0)        
 crayon                 1.3.4     2017-09-16 CRAN (R 3.5.0)        
 crosstalk              1.0.0     2016-12-21 CRAN (R 3.5.0)        
 data.table             1.11.4    2018-05-27 CRAN (R 3.5.0)        
 datasets             * 3.5.1     2018-07-05 local                 
 DBI                    1.0.0     2018-05-02 CRAN (R 3.5.0)        
 DelayedArray         * 0.6.3     2018-08-01 Bioconductor          
 DelayedMatrixStats     1.2.0     2018-05-01 Bioconductor          
 devtools               1.13.6    2018-06-27 CRAN (R 3.5.0)        
 digest                 0.6.15    2018-01-28 CRAN (R 3.5.0)        
 dplyr                * 0.7.6     2018-06-29 CRAN (R 3.5.1)        
 DT                     0.4       2018-01-30 CRAN (R 3.5.0)        
 edgeR                * 3.22.3    2018-06-21 Bioconductor          
 evaluate               0.11      2018-07-17 CRAN (R 3.5.0)        
 fansi                  0.2.3     2018-05-06 CRAN (R 3.5.0)        
 fdrtool                1.2.15    2015-07-08 CRAN (R 3.5.0)        
 forcats              * 0.3.0     2018-02-19 CRAN (R 3.5.0)        
 gdtools              * 0.1.7     2018-02-27 CRAN (R 3.5.0)        
 GenomeInfoDb         * 1.16.0    2018-05-01 Bioconductor          
 GenomeInfoDbData       1.1.0     2018-04-25 Bioconductor          
 GenomicRanges        * 1.32.6    2018-07-20 Bioconductor          
 ggbeeswarm             0.6.0     2017-08-07 CRAN (R 3.5.0)        
 ggcorrplot           * 0.1.1     2016-01-12 CRAN (R 3.5.0)        
 ggforce              * 0.1.3     2018-07-07 CRAN (R 3.5.0)        
 ggplot2              * 3.0.0     2018-07-03 CRAN (R 3.5.0)        
 ggrepel              * 0.8.0     2018-05-09 CRAN (R 3.5.0)        
 ggridges             * 0.5.0     2018-04-05 CRAN (R 3.5.0)        
 ggthemes               4.0.0     2018-07-19 CRAN (R 3.5.0)        
 git2r                  0.23.0    2018-07-17 CRAN (R 3.5.0)        
 glue                   1.3.0     2018-07-17 CRAN (R 3.5.0)        
 graphics             * 3.5.1     2018-07-05 local                 
 grDevices            * 3.5.1     2018-07-05 local                 
 grid                   3.5.1     2018-07-05 local                 
 gridExtra              2.3       2017-09-09 CRAN (R 3.5.0)        
 gtable                 0.2.0     2016-02-26 CRAN (R 3.5.0)        
 haven                  1.1.2     2018-06-27 CRAN (R 3.5.0)        
 hms                    0.4.2     2018-03-10 CRAN (R 3.5.0)        
 htmltools              0.3.6     2017-04-28 CRAN (R 3.5.0)        
 htmlwidgets            1.2       2018-04-19 CRAN (R 3.5.0)        
 httpuv                 1.4.5     2018-07-19 CRAN (R 3.5.0)        
 httr                   1.3.1     2017-08-20 CRAN (R 3.5.0)        
 IHW                  * 1.8.0     2018-05-01 Bioconductor          
 IRanges              * 2.14.10   2018-05-16 Bioconductor          
 jsonlite               1.5       2017-06-01 CRAN (R 3.5.0)        
 knitr                  1.20      2018-02-20 CRAN (R 3.5.0)        
 labeling               0.3       2014-08-23 CRAN (R 3.5.0)        
 later                  0.7.3     2018-06-08 CRAN (R 3.5.0)        
 lattice                0.20-35   2017-03-25 CRAN (R 3.5.1)        
 lazyeval               0.2.1     2017-10-29 CRAN (R 3.5.0)        
 limma                * 3.36.2    2018-06-21 Bioconductor          
 locfit                 1.5-9.1   2013-04-20 CRAN (R 3.5.0)        
 lpsymphony             1.8.0     2018-05-01 Bioconductor (R 3.5.0)
 lubridate              1.7.4     2018-04-11 CRAN (R 3.5.0)        
 magrittr               1.5       2014-11-22 CRAN (R 3.5.0)        
 MASS                   7.3-50    2018-04-30 CRAN (R 3.5.1)        
 Matrix                 1.2-14    2018-04-13 CRAN (R 3.5.1)        
 matrixStats          * 0.54.0    2018-07-23 CRAN (R 3.5.0)        
 memoise                1.1.0     2017-04-21 CRAN (R 3.5.0)        
 methods              * 3.5.1     2018-07-05 local                 
 mime                   0.5       2016-07-07 CRAN (R 3.5.0)        
 modelr                 0.1.2     2018-05-11 CRAN (R 3.5.0)        
 munsell                0.5.0     2018-06-12 CRAN (R 3.5.0)        
 nlme                   3.1-137   2018-04-07 CRAN (R 3.5.1)        
 org.Hs.eg.db         * 3.6.0     2018-05-15 Bioconductor          
 parallel             * 3.5.1     2018-07-05 local                 
 pillar                 1.3.0     2018-07-14 CRAN (R 3.5.0)        
 pkgconfig              2.0.1     2017-03-21 CRAN (R 3.5.0)        
 plyr                   1.8.4     2016-06-08 CRAN (R 3.5.0)        
 promises               1.0.1     2018-04-13 CRAN (R 3.5.0)        
 purrr                * 0.2.5     2018-05-29 CRAN (R 3.5.0)        
 R.methodsS3            1.7.1     2016-02-16 CRAN (R 3.5.0)        
 R.oo                   1.22.0    2018-04-22 CRAN (R 3.5.0)        
 R.utils                2.6.0     2017-11-05 CRAN (R 3.5.0)        
 R6                     2.2.2     2017-06-17 CRAN (R 3.5.0)        
 RColorBrewer         * 1.1-2     2014-12-07 CRAN (R 3.5.0)        
 Rcpp                   0.12.18   2018-07-23 CRAN (R 3.5.0)        
 RCurl                  1.95-4.11 2018-07-15 CRAN (R 3.5.0)        
 readr                * 1.1.1     2017-05-16 CRAN (R 3.5.0)        
 readxl                 1.1.0     2018-04-20 CRAN (R 3.5.0)        
 reshape2               1.4.3     2017-12-11 CRAN (R 3.5.0)        
 rhdf5                  2.24.0    2018-05-01 Bioconductor          
 Rhdf5lib               1.2.1     2018-05-17 Bioconductor          
 rjson                  0.2.20    2018-06-08 CRAN (R 3.5.0)        
 rlang                * 0.2.1     2018-05-30 CRAN (R 3.5.0)        
 rmarkdown              1.10      2018-06-11 CRAN (R 3.5.0)        
 rprojroot              1.3-2     2018-01-03 CRAN (R 3.5.0)        
 RSQLite                2.1.1     2018-05-06 CRAN (R 3.5.0)        
 rstudioapi             0.7       2017-09-07 CRAN (R 3.5.0)        
 rvest                  0.3.2     2016-06-17 CRAN (R 3.5.0)        
 S4Vectors            * 0.18.3    2018-06-08 Bioconductor          
 scales                 0.5.0     2017-08-24 CRAN (R 3.5.0)        
 scater               * 1.9.12    2018-08-03 Bioconductor          
 shiny                  1.1.0     2018-05-17 CRAN (R 3.5.0)        
 SingleCellExperiment * 1.2.0     2018-05-01 Bioconductor          
 slam                   0.1-43    2018-04-23 CRAN (R 3.5.0)        
 stats                * 3.5.1     2018-07-05 local                 
 stats4               * 3.5.1     2018-07-05 local                 
 stringi                1.2.4     2018-07-20 CRAN (R 3.5.0)        
 stringr              * 1.3.1     2018-05-10 CRAN (R 3.5.0)        
 SummarizedExperiment * 1.10.1    2018-05-11 Bioconductor          
 superheat            * 0.1.0     2017-02-04 CRAN (R 3.5.0)        
 svglite                1.2.1     2017-09-11 CRAN (R 3.5.0)        
 tibble               * 1.4.2     2018-01-22 CRAN (R 3.5.0)        
 tidyr                * 0.8.1     2018-05-18 CRAN (R 3.5.0)        
 tidyselect             0.2.4     2018-02-26 CRAN (R 3.5.0)        
 tidyverse            * 1.2.1     2017-11-14 CRAN (R 3.5.0)        
 tools                  3.5.1     2018-07-05 local                 
 tweenr                 0.1.5     2016-10-10 CRAN (R 3.5.0)        
 tximport               1.8.0     2018-05-01 Bioconductor          
 units                  0.6-0     2018-06-09 CRAN (R 3.5.0)        
 utf8                   1.1.4     2018-05-24 CRAN (R 3.5.0)        
 utils                * 3.5.1     2018-07-05 local                 
 vipor                  0.4.5     2017-03-22 CRAN (R 3.5.0)        
 viridis              * 0.5.1     2018-03-29 CRAN (R 3.5.0)        
 viridisLite          * 0.3.0     2018-02-01 CRAN (R 3.5.0)        
 whisker                0.3-2     2013-04-28 CRAN (R 3.5.0)        
 withr                  2.1.2     2018-03-15 CRAN (R 3.5.0)        
 workflowr              1.1.1     2018-07-06 CRAN (R 3.5.0)        
 xml2                   1.2.0     2018-01-24 CRAN (R 3.5.0)        
 xtable                 1.8-2     2016-02-05 CRAN (R 3.5.0)        
 XVector                0.20.0    2018-05-01 Bioconductor          
 yaml                   2.2.0     2018-07-25 CRAN (R 3.5.1)        
 zlibbioc               1.26.0    2018-05-01 Bioconductor          </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
